docker-setups/angular-docker.md

```diff
@@ -0,0 +1,310 @@
+- [Node.js](https://nodejs.org/en/download/)
+- [Angular CLI](https://angular.io/guide/setup-local)
+- [Visual Studio Code (Optional)](https://code.visualstudio.com/)
+
+[Back to top](#table-of-contents)
+
+## 3. Set up for Angular project
+
+In `package.json`, set the command to build the application with production configuration:
+
+```json
+// package.json
+{
+  // ...
+  "scripts": {
+    // Other Commands
+    // Add the following line under "scripts"
+    "build:prod": "ng build --configuration production"
+  }
+  // ...
+}
+```
+
+[Back to top](#table-of-contents)
+
+## 4. Setting up the Docker Environment
+
+### 4.1 Dockerfile
+
+Create a Dockerfile in the root directory of your Angular application.
+
+```Dockerfile
+### Stage 0: Build ###
+#--------------------------------------
+# Use an official Node runtime as a parent image
+FROM node:20 AS builder
+
+# Set the working directory in the container
+WORKDIR /usr/src/app
+
+# Copy package.json and package-lock.json to the working directory
+COPY package*.json ./
+
+# Install Angular CLI Globally
+RUN npm install -g @angular/cli
+
+# Install app dependencies
+RUN npm install
+
+# Copy the rest of the application code
+COPY . .
+
+# Build the Angular app for production
+RUN npm run build:prod
+
+### Stage 1: Run ###
+#--------------------------------------
+# Defining nginx image to be used
+FROM nginx:latest AS ngi
+
+# Fetch APP_NAME from docker-compose
+ARG APP_NAME
+
+# Create App Path for build files for Angular 17 and above projects
+# For versions older than 17, replace APP_PATH with /usr/src/app/dist/*
+ENV APP_PATH=/usr/src/app/dist/${APP_NAME}/browser
+
+# Copy the built Angular app to the default Nginx public folder
+COPY --from=builder $APP_PATH /usr/share/nginx/html/
+
+# Need to make nginx config file
+COPY nginx.conf /etc/nginx/conf.d/default.conf
+
+# Expose port 80
+EXPOSE 80
+
+# Start Nginx when the container starts
+CMD ["nginx", "-g", "daemon off;"]
+```
+
+### Set the Correct File Path for Nginx
+
+#### Why?
+
+- When you run the command `ng build`, a `dist` folder is created which contains all angular build files.
+- Depending on the CLI version, different folder structures are created.
+- We need to give the path of `index.html` for Nginx to work.
+
+#### How to find folder name?
+
+- Folder name is present in `package.json` as `"name": "your-app-name"`
+- Verify the folder name by running `ng build` and checking in `dist` folder.
+
+#### 1. For angular version 17
+
+```
+build file folder structure
+
+dist
+| your-app-name
+| | browser
+| | | index.html
+| | | <...other build files>
+| | license
+```
+**Define APP_NAME in `docker-compose.yml`:**
+
+- Define `${APP_NAME}` in [docker compose file](#42-docker-compose) and skip this section.
+
+**OR ALTERNATIVELY**
+
+**Update `Dockerfile`:**
+
+- Update `ENV APP_PATH` by directly replacing `${APP_NAME}` with your application name like so:
+> ENV APP_PATH=/usr/src/app/dist/`your-app-name`/browser
```

Comment:
Can replace this quoteblock with code block like this:
````
```
ENV APP_PATH=/usr/src/app/dist/`your-app-name`/browser
```
````

Reply:
Used quoteblocks